src/App.tsx

```diff
@@ -15,47 +15,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE = 0,
+  ALPHABET = 1,
+  LENGTH = 2,
+}
+
+function getReorderedGoods(
+  goods: string[],
+  isReversed: boolean,
+  sortType: SortType,
+):string[] {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((goodFirst, goodSecond) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return goodFirst.localeCompare(goodSecond);
+      case SortType.LENGTH:
+        return goodFirst.length - goodSecond.length;
+      case SortType.NONE:
+      default:
+        return 0;
+    }
+  });
+
+  return isReversed ? visibleGoods.reverse() : visibleGoods;
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const copiedData = [...goodsFromServer];
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const reverse = () => {
+    setIsReversed(prevState => !prevState);
+  };
+
+  const onReset = () => {
```

Comment:
```suggestion
  const handleReset = () => {
```
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364